R/mig_calculate_rc.R

```diff
@@ -24,21 +23,37 @@
 #' alpha2= 0.1, mu2= 21, lambda2= 0.39, a3= 0.001, 
 #' alpha3= 1, mu3= 67, lambda3= 0.6, c= 0.01)
 #' ages <- 0:75
-#' mx <- mig_calculate_rc(ages = ages, pars = pars, num_pars = 11)
+#' mx <- mig_calculate_rc(ages = ages, pars = pars)
 #' plot(ages, mx, type = 'l')
 
 mig_calculate_rc <- function(ages,
-                             pars,
-                             num_pars){
+                             pars){
   
-  # simple check. Really I guess specific combos are the important thing
+  # parameter name groups
+  comp1 <- c("a1", "alpha1", "c")
+  comp2 <- c("a2", "alpha2", "lambda2", "mu2")
+  comp3 <- c("a3", "alpha3", "lambda3", "mu3")
+  comp4 <- c("a4", "lambda4")
+  
+  
+  # simple check
   stopifnot(length(pars) %in% c(7, 9, 11, 13))
+  # check for specific parameter groups
+  if(length(pars)==7)
+    stopifnot(all(c(comp1, comp2) %in% names(pars)))
+  if(length(pars)==9)
+    stopifnot(all(c(comp1, comp2, comp4) %in% names(pars)))
+  if(length(pars)==11)
+    stopifnot(all(c(comp1, comp2, comp3) %in% names(pars)))
+  if(length(pars)==13)
+    stopifnot(all(c(comp1, comp2, comp3, comp4) %in% names(pars)))
```

Comment:
# TR are these really all valid combos? Is c(comp1, comp3) not possible? If that's so, then checks along these lines could suffice:
```
if (any(comp1 %in% names(pars)){
stopifnot(all(comp1 %in% names(pars)))
}
```

Reply:
# And `c` is give-or-take?

Reply:
yes. good point. I was just doing the usual combos but in theory you could do any combo. will update.